### PR TITLE
Feature/register

### PR DESCRIPTION
**Pull request by caesargusti**
model/ customers
model/config
api/customers/customers
api/router

### CODE REVIEW

File: models/customers.go [hunk 1]
```diff
@@ -52,3 +88,4 @@ func GetProfile(db *sql.DB, id int) (Customers, error) {
 	}
 	return cus, nil
 }
+>>>>>>> 7a28cdc18787521684b684f1a16e0f68de1795e1
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

File: models/customers.go [hunk 2]
```diff
@@ -36,11 +68,15 @@ func RegisterCustomer(db *sql.DB, objCustomer Customers, AccNum string) error {
...
 	return err
+>>>>>>> 7a28cdc18787521684b684f1a16e0f68de1795e1
 }
 
 func LoginCustomer(db *sql.DB, email string, password string) (objCustomer Customers, err error) {
 	err = db.QueryRow("SELECT cust_id, account_num, cust_name, cust_address, cust_phone, cust_email, cust_password, is_verified, channel, created_at, updated_at from customers where is_verified = true and cust_email = ($1) and cust_password = ($2)", email, password).Scan(&objCustomer.CustId, &objCustomer.AccountNum, &objCustomer.CustName, &objCustomer.CustAddress, &objCustomer.CustPhone, &objCustomer.CustEmail, &objCustomer.CustPassword, &objCustomer.IsVerified, &objCustomer.Channel, &objCustomer.CreatedAt, &objCustomer.UpdatedAt)
 	return
+<<<<<<< HEAD
+}
+=======
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

File: models/customers.go [hunk 3]
```diff
@@ -36,11 +68,15 @@ func RegisterCustomer(db *sql.DB, objCustomer Customers, AccNum string) error {
 		Update,
 	)
 	return err
+>>>>>>> 7a28cdc18787521684b684f1a16e0f68de1795e1
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

File: models/customers.go [hunk 4]
```diff
@@ -21,10 +39,24 @@ type Customers struct {
...
+			objCustomer.CustPhone,
+			objCustomer.CustEmail,
+			Pass,
+			Verified,
+			objCustomer.Channel,
+			Create,
+			Update,
+		)
+		return err
+=======
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

File: models/customers.go [hunk 5]
```diff
@@ -21,10 +39,24 @@ type Customers struct {
 }
 
 func RegisterCustomer(db *sql.DB, objCustomer Customers, AccNum string) error {
+>>>>>>> 7a28cdc18787521684b684f1a16e0f68de1795e1
 	Create := time.Now()
 	Update := time.Now()
 	Verified := false
 	_, err := db.Exec("INSERT into customers (account_num, cust_name, cust_address, cust_phone, cust_email, cust_password, is_verified, channel, created_at, updated_at) values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)", AccNum,
+<<<<<<< HEAD
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

File: models/customers.go [hunk 6]
```diff
@@ -21,10 +39,24 @@ type Customers struct {
...
+			objCustomer.CustAddress,
+			objCustomer.CustPhone,
+			objCustomer.CustEmail,
+			Pass,
+			Verified,
+			objCustomer.Channel,
+			Create,
+			Update,
+		)
+		return err
```
Comment (vicimilenia) [inline models/customers.go:39]:
```suggestion
		return
```

File: api/customers/customers.go [hunk 7]
```diff
@@ -1,22 +1,86 @@
...
+
+	if len(cus.CustPassword) < 6 {
+		helpers.HTTPError(w, http.StatusBadRequest, "Password Min 6 Character")
+		return
+	}
+		
+	date := time.Now()
+	now := date.Format("060102")
+	rand.Seed(time.Now().UnixNano())
+	randomNumber := rand.Intn(9999)
```
Comment (polipopoliko) [inline api/customers/customers.go:52]: Hmm.. thinking..

**Review by vicimilenia — APPROVED** [review]

File: models/customers.go [hunk 8]
```diff
@@ -6,6 +6,24 @@ import (
...
+	CustPict		string		`json:"cust_pict"`
+	IsVerified 		bool 		`json:"is_verified"`
+	Channel 		string 		`json:"channel"`
+	CreatedAt 		time.Time 	`json:"created_at"`
+	UpdatedAt 		time.Time 	`json:"updated_at"`
+}
+
+
+func RegisterCustomer(db *sql.DB, objCustomer Customers, AccNum string, Pass string) error {
+=======
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

**Review by polipopoliko — COMMENTED** [review]
IF there's a spare time we should change it to avoid unique constraint

File: models/customers.go [hunk 9]
```diff
@@ -21,10 +39,24 @@ type Customers struct {
 }
 
 func RegisterCustomer(db *sql.DB, objCustomer Customers, AccNum string) error {
+>>>>>>> 7a28cdc18787521684b684f1a16e0f68de1795e1
```
Comment (vicimilenia) [inline models/customers.go]: please delete this

**Review by vicimilenia — CHANGES_REQUESTED** [review]

**Pull request merged**